### PR TITLE
Use config vars instead of env vars

### PR DESCRIPTION
**Pull request by Gamah**
Felt compelled to clean this up, if you're interested.

### CODE REVIEW

**Comment by matthieugrieger** [comment]
Thanks for the PR! This is actually already in the works. :)

I'm currently doing a rewrite/refactor of MumbleDJ, here is how it is handled now: https://github.com/matthieugrieger/mumbledj/blob/refactor/dj/config.go#L39-L40

Unfortunately merging this change would break all MumbleDJ installations and require users to migrate their API keys to the config file. I would like to avoid that.

Thanks for the PR though! It's definitely a good idea and probably a better way to handle the API key configuration. Environment variables are a bit of a hassle.

**Comment by Gamah** [comment]
I wouldn't be against modifying the PR and scripting the migration of existing env vars to the new config file if you could be convinced to merge that. If not, no worries. I had a good time really digging through a golang project for the first time anyway. :)

**Comment by matthieugrieger** [comment]
I think it would be easier to just check for both an environment variable and a config value and use whichever one exists.

If you want to make the change and update the PR, go ahead and I'll merge it. Just know that it will go away fairly soon when I finish the refactor. :)

**Comment by Gamah** [comment]
You'll have to forgive me, I'm pretty terrible with git. I wrote and tested a quick sed expression which mitigates your concerns with existing installs, my branch reflects this but for some reason it's not showing in this PR? I suspect because it is closed?

Either way, I can't think yet of a case where this should break. I have considered:
* New installs
    * Readme explains where to put the keys
* Existing installs retaining env vars
    * sed expression updates the config file on make install
* Existing installs who are updating sans env vars
    * Covered by existing config file backup logic

**Comment by matthieugrieger** [comment]
I do see your changes in your fork. I'll try reopening the issue to see if that will bring in the changes and update the PR.

**Comment by matthieugrieger** [comment]
Alright, the changes are now in this PR. Give me some time to test this change and I will merge it if it works.

For clarification, could you explain what you mean by your third point of consideration? Is this talking about those who have already migrated their env vars to the config file and are updating MumbleDJ afterwards?

**Comment by Gamah** [comment]
I suppose I tried too hard to explain the third point. The gist of it is if there's no env vars, the new config file will be generated with neither key set, but of course this is only after backing up the existing file.

The only issue I could predict is someone removing their API key env vars before first pulling down these changes and running a make install, though I'd argue any user competent enough to understand the changes well enough to do that is probably also equipped enough to go ahead and move them to their config file as well.

tl;dr: Don't remove your env vars unless you know what you're doing.

I appreciate your reconsideration, and will now poke around the refactoring branch too! Golang looks interesting and this is a lovely project.

**Comment by matthieugrieger** [comment]
Ah, I see. I'll take a look at it later. I am going on a trip this weekend so it will be a few days until I can get to it.

**Comment by Gamah** [comment]
Making a note for myself for later, in case I get to it before you merge:

There are a few "static" dependencies which this relies on which could (should?) be checked or in some way allowed to be checked to be sure they are up to date. Namely youtube_dl (which broke for me tonight, though mumbledj did not crash!) It stopped pulling audio files down because youtube made a change, which required youtube_dl to be updated (thanks pip!)

**Comment by matthieugrieger** [comment]
That's another thing that I have been planning to address in the refactor.

Unfortunately I've gotten sick so it will still be a few more days until I get to anything MumbleDJ-related.

**Comment by Gamah** [comment]
Cheers, take your time!

**Comment by raz0rsdge** [comment]
I wish you all the best... hopefully you find some time to recover your health.

**Comment by matthieugrieger** [comment]
Status update...

I am not sick anymore, but now I am in the midst of university finals. There will be some more delay unfortunately. But once the exams are over I should have a lot of time!

**Comment by Gamah** [comment]
Cheers, best of luck on those!

**Comment by matthieugrieger** [comment]
Finally got around to testing this. Seems to work great, thanks again!
